flink-java/src/main/java/org/apache/flink/api/java/operators/SingleInputUdfOperator.java

```diff
@@ -54,8 +54,11 @@
 
 	private Map<String, DataSet<?>> broadcastVariables;
 
+	// NOTE: only set this variable via setSemanticProperties()
```

Comment:
Manual annotations should always trump optimizer annotations. The analyzer can not determine all semantic properties. E.g. when using KeySelectors. The user should still have the possibility to override semantic properties to add more properties.